doc/conf.py

```diff
@@ -210,15 +210,30 @@ def matplotlib_reduced_latex_scraper(block, block_vars, gallery_conf,
 mathmpl_fontsize = 11.0
 mathmpl_srcset = ['2x']
 
-# Monkey-patching gallery signature to include search keywords
-gen_rst.SPHX_GLR_SIG = """\n
+# Monkey-patching gallery header to include search keywords
+gen_rst.EXAMPLE_HEADER = """
+.. DO NOT EDIT.
+.. THIS FILE WAS AUTOMATICALLY GENERATED BY SPHINX-GALLERY.
+.. TO MAKE CHANGES, EDIT THE SOURCE PYTHON FILE:
+.. "{0}"
+.. LINE NUMBERS ARE GIVEN BELOW.
+
 .. only:: html
 
- .. rst-class:: sphx-glr-signature
+    .. meta::
+        :keywords: matplotlib code example, codex, python plot, pyplot
```

Comment:
Ah, so the keywords come from a template and are the same for all gallery pages. That makes them relatively unhelpful. I think we can remove everything but 'codex'

```suggestion
        :keywords: codex
```

Reply:
Maybe?  "Codex matplotlib ellipse" and "code example matplotlib ellipse" seem identical in search.  Not sure if "code example matplotlib" would work as well?   "Gallery matplotlib ellipse" also works.  "Matplotlib ellipse" returns the method, so there is some benefit to adding "example" in the kwargs I think.

Reply:
I'm a bit confused what are keywords and what are example search strings in your answer.

Also, can somebody who understands the topic please clarify what the effect of these keywords is. Is that just some strings we put intro the site so that search can find these strings somewhere? Is this only for the internal search or is this targeted at external search engines as well?

Reply:
I was just talking about google search terms. OTOH it seems that google ignores the keyword tag, so maybe this is not worth doing, or maybe we should explicitly include as we were before.  

I have no idea how the internal search works.

Reply:
Yes, google does not use the keywords meta tag. If we want to add it, that should be for our internal purposes. And the only internal keyword purpose I see is the "codex" flag. Therefore my suggestion to delete everything else, we don't need to communicate to our internal search that we're matplotlib or about python plotting. The only thing I know about the internal search is that it's javascript that somehow parses the html sources. Empirically, this PR shows that "codex" is still found in the keywords meta tag. That's why I suggested to only keep "codex".